tools/clang/test/HLSLFileCheck/d3dreflect/mesh-node-rdat.hlsl

```diff
@@ -0,0 +1,398 @@
+// CHECK:              SigOutputElements: <{{[0-9]+}}:RecordArrayRef<SignatureElement>[1]>  = {
+// CHECK:                [0]: <0:SignatureElement> = {
+// CHECK:                  SemanticName: "SV_Position"
+// CHECK:                  SemanticIndices: <8:array[1]> = { 0 }
+// CHECK:                  SemanticKind: Position
+// CHECK:                  ComponentType: F32
+// CHECK:                  InterpolationMode: LinearNoperspective
+// CHECK:                  StartRow: 0
+// CHECK:                  ColsAndStream: 3
+// CHECK:                  UsageAndDynIndexMasks: 0
+// CHECK:                }
+// CHECK:              }
+// CHECK:              SigPrimOutputElements: <{{[0-9]+}}:RecordArrayRef<SignatureElement>[1]>  = {
+// CHECK:                [0]: <1:SignatureElement> = {
+// CHECK:                  SemanticName: "CLR"
+// CHECK:                  SemanticIndices: <8:array[1]> = { 0 }
+// CHECK:                  SemanticKind: Arbitrary
+// CHECK:                  ComponentType: F32
+// CHECK:                  InterpolationMode: Constant
+// CHECK:                  StartRow: 0
+// CHECK:                  ColsAndStream: 3
+// CHECK:                  UsageAndDynIndexMasks: 0
+// CHECK:                }
+// CHECK:              }
+// CHECK:              ViewIDOutputMask: <0:bytes[0]>
+// CHECK:              ViewIDPrimOutputMask: <0:bytes[0]>
+// CHECK:              NumThreads: <0:array[3]> = { 1, 1, 1 }
+// CHECK:              GroupSharedBytesUsed: 0
+// CHECK:              GroupSharedBytesDependentOnViewID: 0
+// CHECK:              PayloadSizeInBytes: 0
+// CHECK:              MaxOutputVertices: 3
+// CHECK:              MaxOutputPrimitives: 1
+// CHECK:              MeshOutputTopology: 2
+// CHECK:            }
+// CHECK:          }
+// CHECK:        }
+// CHECK:        Outputs: <RecordArrayRef<IONode>[0]> = {}
+// CHECK:        Inputs: <{{[0-9]+}}:RecordArrayRef<IONode>[1]>  = {
+// CHECK:          [0]: <0:IONode> = {
+// CHECK:            IOFlagsAndKind: 97
+// CHECK:            Attribs: <{{[0-9]+}}:RecordArrayRef<NodeShaderIOAttrib>[3]>  = {
+// CHECK:              [0]: <0:NodeShaderIOAttrib> = {
+// CHECK:                AttribKind: RecordSizeInBytes
+// CHECK:                RecordSizeInBytes: 12
+// CHECK:              }
+// CHECK:              [1]: <1:NodeShaderIOAttrib> = {
+// CHECK:                AttribKind: RecordDispatchGrid
+// CHECK:                RecordDispatchGrid: <RecordDispatchGrid>
+// CHECK:                  ByteOffset: 0
+// CHECK:                  ComponentNumAndType: 23
+// CHECK:              }
+// CHECK:              [2]: <2:NodeShaderIOAttrib> = {
+// CHECK:                AttribKind: RecordAlignmentInBytes
+// CHECK:                RecordAlignmentInBytes: 4
+// CHECK:              }
+// CHECK:            }
+// CHECK:          }
+// CHECK:        }
+// CHECK:      }
+// CHECK:    }
+// CHECK:  }
+// CHECK:  RecordTable (stride = {{[0-9]+}} bytes) NodeIDTable[1] = {
```

Comment:
I don't think we need checks for all the remaining RDAT tables, since all the details are expanded in-line the first time they are used in the `FunctionTable`.